src/Compilers/CSharp/Test/Emit/Emit/EmitMetadataTests.cs

```diff
@@ -2936,5 +2936,546 @@ public class Child : Parent, IParent
+            comp.MakeMemberMissing(WellKnownMember.System_Text_Encoding__get_UTF8);
+            comp.MakeMemberMissing(WellKnownMember.System_Text_Encoding__GetString);
+            comp.VerifyEmitDiagnostics(
+                // (2,26): error CS0656: Missing compiler required member 'System.Text.Encoding.get_UTF8'
+                // System.Console.Write("bb");
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""bb""").WithArguments("System.Text.Encoding", "get_UTF8").WithLocation(2, 22),
+                // (2,26): error CS0656: Missing compiler required member 'System.Text.Encoding.GetString'
+                // System.Console.Write("bb");
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""bb""").WithArguments("System.Text.Encoding", "GetString").WithLocation(2, 22),
+                // (3,22): error CS0656: Missing compiler required member 'System.Text.Encoding.get_UTF8'
+                // System.Console.Write("ccc");
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""ccc""").WithArguments("System.Text.Encoding", "get_UTF8").WithLocation(3, 22),
+                // (3,22): error CS0656: Missing compiler required member 'System.Text.Encoding.GetString'
+                // System.Console.Write("ccc");
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""ccc""").WithArguments("System.Text.Encoding", "GetString").WithLocation(3, 22));
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.Regular.WithFeature("experimental-data-section-string-literals", "3"));
+            comp.MakeMemberMissing(WellKnownMember.System_Text_Encoding__get_UTF8);
+            comp.MakeMemberMissing(WellKnownMember.System_Text_Encoding__GetString);
+            CompileAndVerify(comp, expectedOutput: "abbccc").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DataSectionStringLiterals_Threshold()
+        {
+            var source = """
+                System.Console.Write("a");
+                System.Console.Write("bb");
+                System.Console.Write("ccc");
+                """;
+
+            var expectedOutput = "abbccc";
+
+            var expectedIl = """
+                {
+                  // Code size       31 (0x1f)
+                  .maxstack  1
+                  IL_0000:  ldstr      "a"
+                  IL_0005:  call       "void System.Console.Write(string)"
+                  IL_000a:  ldstr      "bb"
+                  IL_000f:  call       "void System.Console.Write(string)"
+                  IL_0014:  ldstr      "ccc"
+                  IL_0019:  call       "void System.Console.Write(string)"
+                  IL_001e:  ret
+                }
+                """;
+
+            CompileAndVerify(source, expectedOutput: expectedOutput)
+                .VerifyDiagnostics()
+                .VerifyIL("<top-level-statements-entry-point>", expectedIl);
```

Comment:
>VerifyIL [](http://example.com/codeflow?start=17&length=8)

Unfortunately, here and in the loop below, verifying IL doesn't feel sufficient enough. For example, if, for whatever reason, the ```DataSectionStringLiteralThreshold``` would be set to >=3, we wouldn't be able to observe this fact. What our ultimate goal here is to confirm whether the feature gets "activated" and, if so, with what threshold. I suggest explicitly asserting expected value of ```DataSectionStringLiteralThreshold``` for each of these scenarios.